scmrepo/git/backend/pygit2.py

```diff
@@ -3,6 +3,7 @@
 import os
 import stat
 from contextlib import contextmanager
+from functools import cached_property
```

Comment:
I'll suggest sticking to `funcy.cached_property` instead due to https://github.com/python/cpython/issues/87634.

Unless we are trying to replace the following piece of code:
```python
@funcy.wrap_prop(threading.RLock())
@funcy.cached_property

# equivalent to @functools.cached_property
```

Reply:
Makes sense, was just trying it out with pylint and 3.8

Reply:
Didn't know about that bug though 🙁 Thank you!